code/__DEFINES/waddle.dm

```diff
@@ -0,0 +1,14 @@
+#define ADD_WADDLE(target, source) if(!HAS_TRAIT(target, TRAIT_WADDLING))target.AddElement(/datum/element/waddling);ADD_TRAIT(target, TRAIT_WADDLING, source)
+#define REMOVE_WADDLE(target, source) REMOVE_TRAIT(target, TRAIT_WADDLING, source); if(!HAS_TRAIT(target, TRAIT_WADDLING)) target.RemoveElement(/datum/element/waddling)
```

Comment:
these defines are completely unnecessary, just hook a signal to add/remove TRAIT_WADDLING to add/remove the waddling element

Reply:
that is correct

Reply:
this define is dogshit but youre adamant on not using registersignal because youre a moron so whatever